Gems/SceneProcessing/Code/Source/Config/Components/SceneProcessingConfigSystemComponent.cpp

```diff
@@ -39,23 +39,56 @@ namespace AZ
-            // Defaults in case there's no config setup
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]1(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh1", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]2(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh2", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]3(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh3", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]4(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh4", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]5(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh5", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Pp][Hh][Yy][Ss](_optimized)?$", PatternMatcher::MatchApproach::Regex, "PhysicsMesh", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("_ignore", PatternMatcher::MatchApproach::PostFix, "Ignore", false));
-            // If the filename ends with "_anim" this will mark all nodes as "Ignore" unless they're derived from IAnimationData. This will
-            // cause only animations to be exported from the source scene file even if there's other data available.
-            m_softNames.push_back(aznew FileSoftNameSetting("_anim", PatternMatcher::MatchApproach::PostFix, "Ignore", false,
-                { FileSoftNameSetting::GraphType(SceneAPI::DataTypes::IAnimationData::TYPEINFO_Name()) }));
 
+            PopulateDefaultSoftNameSettings();
             m_UseCustomNormals = true;
         }
 
+        void SceneProcessingConfigSystemComponent::PopulateDefaultSoftNameSettings()
+        {
+            auto settingsRegistry = AZ::SettingsRegistry::Get();
+            if (settingsRegistry)
+            {
+                AZStd::vector<NodeSoftNameSetting*> nodeSoftNameSettings;
+                settingsRegistry->GetObject(nodeSoftNameSettings, AssetProcessorDefaultNodeSoftNameSettingsKey);
+                for (NodeSoftNameSetting* nodeSoftNameSetting : nodeSoftNameSettings)
+                {
+                    if (!AddSoftName(nodeSoftNameSetting))
+                    {
+                        delete nodeSoftNameSetting;
+                        nodeSoftNameSetting = nullptr;
+                    }
+                }
+
+                AZStd::vector<FileSoftNameSetting*> fileSoftNameSettings;
+                settingsRegistry->GetObject(fileSoftNameSettings, AssetProcessorDefaultFileSoftNameSettingsKey);
+                for (FileSoftNameSetting* fileSoftNameSetting : fileSoftNameSettings)
+                {
+                    if (!AddSoftName(fileSoftNameSetting))
+                    {
+                        delete fileSoftNameSetting;
+                        fileSoftNameSetting = nullptr;
+                    }
+                }
```

Comment:
Instead of doing this manual memory management, I recommend changing the container to use a smart pointer or just storing the object directly in the container. The container can manage its own memory. Nothing is checking for pointer validity anyway.

```suggestion
                AZStd::vector<NodeSoftNameSetting*> nodeSoftNameSettings;
                settingsRegistry->GetObject(nodeSoftNameSettings, AssetProcessorDefaultNodeSoftNameSettingsKey);
                for (NodeSoftNameSetting*& nodeSoftNameSetting : nodeSoftNameSettings)
                {
                    if (!AddSoftName(nodeSoftNameSetting))
                    {
                        delete nodeSoftNameSetting;
                        nodeSoftNameSetting = nullptr;
                    }
                }

                AZStd::vector<FileSoftNameSetting*> fileSoftNameSettings;
                settingsRegistry->GetObject(fileSoftNameSettings, AssetProcessorDefaultFileSoftNameSettingsKey);
                for (FileSoftNameSetting*& fileSoftNameSetting : fileSoftNameSettings)
                {
                    if (!AddSoftName(fileSoftNameSetting))
                    {
                        delete fileSoftNameSetting;
                        fileSoftNameSetting = nullptr;
                    }
                }
```

Reply:
[m_softNames](https://github.com/o3de/o3de/blob/development/Gems/SceneProcessing/Code/Source/Config/Components/SceneProcessingConfigSystemComponent.h#L95) is stored as a vector of raw pointers and we're currently managing the memory manually when [clear the vector](https://github.com/o3de/o3de/blob/development/Gems/SceneProcessing/Code/Source/Config/Components/SceneProcessingConfigSystemComponent.cpp#L87-L90). If I change the nodeSoftNameSettings and fileSoftNameSettings to use a smart pointer, elements from them will be deleted when they are out of scope, then m_softNames will include dangling pointers and causes potential issues. I think we may consider to convert m_softNames to use smart pointers in future PRs to fully address this.

Reply:
Updated m_softNames to use unique pointers.